p2p/net/conn.go

```diff
@@ -192,7 +197,34 @@ func (c *FormattedConnection) publish(message []byte) {
 //	return cancel
 //}
 
+func (c *FormattedConnection) sendListener() {
+	for {
+		select {
+		case buf := <-c.messages:
+			//todo: we are hiding the error here...
+			err := c.SendSock(buf)
```

Comment:
You are making it worse than it was.
Before, we were sending on n sockets concurrently and waiting for the slowest.
Now, you are sending on only one socket concurrently and as long as the buffer is not full, we are not waiting at all.
So we have two issues here:
1. We are not using more than one socket concurrently.
2. When the buffer will be full we will do the same/worse - block until we are done sending enough message from the buffer.